src/input/item.rs

```diff
@@ -77,6 +77,170 @@ impl TextItem {
     }
 }
 
+fn is_valid_domain_name(domain: &str) -> bool {
```

Comment:
@octoaide, I was wondering if these checks are sufficient to validate domain names. Are there any additional constraints or rules we should consider?